addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -649,6 +653,146 @@ QUnit.test('basic rendering of canceled notification', async function (assert) {
     );
 });
 
+QUnit.test('delete attachment should delete all text content free messages linked to it', async function (assert) {
+    assert.expect(2);
+
+    this.data['ir.attachment'].records.push({
+        id: 143,
+        mimetype: 'text/plain',
+        name: "Blah.txt",
+    });
+    this.data['mail.channel'].records.push({ id: 11 });
+    this.data['mail.message'].records.push(
+        {
+            attachment_ids: [143],
+            body: "",
+            channel_ids: [11],
+            id: 101,
+        }, {
+            attachment_ids: [143],
+            channel_ids: [11],
+            id: 102,
+        }
+    );
+    await this.start();
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['insert', { id: 11, model: 'mail.channel' }]],
+    });
+    // wait for messages of the thread to be loaded
+    await this.afterEvent({
+        eventName: 'o-component-message-list-thread-cache-changed',
+        func: () => {
+            this.createThreadViewComponent(threadViewer.threadView);
+        },
+        message: "thread become loaded with messages",
+        predicate: ({ threadViewer }) => {
+            return threadViewer.thread.model === 'mail.channel' && threadViewer.thread.id === 11;
+        },
+    });
+
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        2,
+        "there should be 2 messages displayed"
+    );
+    await afterNextRender(() => {
+        document.querySelector(`.o_Attachment[data-attachment-local-id="${
+            this.env.models['mail.attachment'].findFromIdentifyingData({ id: 143 }).localId
+        }"] .o_Attachment_asideItemUnlink`).click();
+    });
+    await afterNextRender(() =>
+        document.querySelector('.o_AttachmentDeleteConfirmDialog_confirmButton').click()
+    );
+    assert.containsNone(
+        document.body,
+        '.o_Message',
+        "there should be no message left displayed"
+    );
+});
+
+QUnit.test('delete attachment should not delete non-empty message linked to it', async function (assert) {
```

Comment:
_"delete all attachments of a message with some text content should still keep it displayed"_